### DISCUSSION TITLE
Regexmanager rejects valid registry urls

### DISCUSSION THREAD

**Discussion by henrysachs** [renovatebot/renovate#10305] [Request Help] [answered]
**How are you running Renovate?**

- [ ] WhiteSource Renovate hosted app on github.com
- [x] Self hosted

If using the hosted app, please skip to the next section.
Otherwise, if self-hosted, please complete the following:

Please select which platform you are using:

- [ ] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server
- [ ] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server
- [ ] Gitea
- [ ] github.com
- [ ] GitHub Enterprise Server
- [ ] gitlab.com
- [x] GitLab self-hosted

Renovate version: latest

**Describe the bug**

I have the following regex to match a docker image:

```
[iI]mage: \\{\\s.*?repository:\\s*\\'(?<registryUrl>.*?)\\/(?<depName>.*?)\\',\\s.*?tag: \\'(?<currentValue>.*?)\\'
```

This is a sample line 

```
              image: {
                repository: 'quay-docker-remote.artifactory.internal.host.redacted.de/prometheus-operator/prometheus-operator',
                tag: 'v0.46.0',
              },
```

this matched my host: quay-docker-remote.artifactory.internal.host.redacted.de as registry url but renovate rejects this url.

**Relevant debug logs**
```
 WARN: Invalid regex manager registryUrl (repository=seec-mop/mop-base/renovate-test-repo)
       "value": "quay-docker-remote.artifactory.internal.host.redacted.de"
```

**Have you created a minimal reproduction repository?**

no This is kinda hard because i think because of the amount of subdomains included the url is rejected

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [x] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

...

**Comment by rarkins** [comment, marked as answer]
You've captured a hostname, and it's not a valid url. Keep it all as part of depName.

**Reply by rarkins** [reply]
Converted to discussion as it is not a bug.

**Reply by henrysachs** [reply]
can you be more explicit about this what would be the registryurl than? would depname include the registry url?

**Reply by rarkins** [reply]
Docker does not need a registry URL. Just use foo.bar.baz/something/else as the depName

**Reply by henrysachs** [reply]
tried this now renovate tries to use docker.io to fetch dependency information I just removed the registryurl part from my regex.

**Reply by rarkins** [reply]
Please create a public reproduction repo so I can see what's going wrong. You can use quay.io or any other public host as the non-Docker.io image instead of your private one

**Reply by henrysachs** [reply]
https://github.com/henrysachs/renovate-test-docker-registry/tree/main

i used non existent hosts. No matter which host i insert there renovate tries to do a fetch on docker.io.

**Reply by rarkins** [reply]
There is no renovate config there

**Reply by rarkins** [reply]
And please reduce it to the simplest example, so that it can be debugged if necessary

**Reply by henrysachs** [reply]
now there is. Thanks for your help.

**Reply by rarkins** [reply]
Sorry, but it's too complicated to run through debug. Please see https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md for what we need before debugging a reproduction

Please also convert it from renovate.js to `renovate.json` - i.e. it should be in-repo config not bot/admin config.

**Reply by henrysachs** [reply]
will try and come back to you. :)

**Reply by henrysachs** [reply]
so i got it too work but it also does the requests to the right repos there. Could it be that there is a mismatch between the js and json file?

**Reply by henrysachs** [reply]
whats also different is that i'm using gitlab. So when using the same config file against a repo in gitlab it has issues.

**Reply by henrysachs** [reply]
yeah so i dont know why but i get different errors when using the gitlab version: https://gitlab.com/DerAstronaut/renovate-test-repo-docker
and the github: 
https://github.com/henrysachs/renovate-test-docker-registry

when using the github one I get the parsed registry correctly. I needed to use the domain scheme that is provided there to get this kind of error.

**Reply by henrysachs** [reply]
oh sorry seems like it parsed them correctly just had different log output. I will try to use the json file in my gitlab instance and see if this changes the output.

**Reply by henrysachs** [reply]
just a heads up it works with the json config! thanks for your help on debugging this not sure if i can get this reproducible publicly because you need an artifactory repo and the javascript config.
